internal/app/cloudinfo/loader/loader.go

```diff
@@ -0,0 +1,301 @@
+				if cloudinfo.Contains(zones.([]string), zone) {
+					availableZones = append(availableZones, zone)
+				}
+			}
+			scil.store.StoreZones(provider, service, region.Id, availableZones)
+		}
+	default:
+		scil.log.Error("invalid zone strategy",
+			map[string]interface{}{"provider": provider, "service": service, "region": region.Id, "strategy": region.Data.Zones.Strategy})
+	}
+}
+
+func (scil *storeCloudInfoLoader) LoadVersions(ctx context.Context, provider string, service string, region Region) {
+	switch region.Data.Versions.Strategy {
+	case exact:
+		scil.log.Debug("loading versions...")
+		scil.store.StoreVersion(provider, service, region.Id, region.Data.Versions.Data)
+		scil.log.Debug("versions loaded")
+	case exclude:
+		var (
+			versions interface{}
+			ok       bool
+		)
+		if versions, ok = scil.store.GetVersion(provider, scil.serviceData.Source, region.Id); !ok {
+			scil.log.Error("versions not yet cached",
+				map[string]interface{}{"provider": provider, "service": scil.serviceData.Source, "region": region.Id})
+		} else {
+			var availableVersions []string
+			for _, version := range versions.([]string) {
+				if !cloudinfo.Contains(region.Data.Versions.Data, version) {
+					availableVersions = append(availableVersions, version)
+				}
+			}
+			scil.store.StoreVersion(provider, service, region.Id, availableVersions)
+		}
+	case include:
+		var (
+			versions interface{}
+			ok       bool
+		)
+		if versions, ok = scil.store.GetVersion(provider, scil.serviceData.Source, region.Id); !ok {
+			scil.log.Error("versions not yet cached",
+				map[string]interface{}{"provider": provider, "service": scil.serviceData.Source, "region": region.Id})
+		} else {
+			var availableVersions []string
+			for _, version := range region.Data.Versions.Data {
+				if cloudinfo.Contains(versions.([]string), version) {
+					availableVersions = append(availableVersions, version)
+				}
+			}
+			scil.store.StoreVersion(provider, service, region.Id, availableVersions)
+		}
+	default:
+		scil.log.Error("invalid version strategy",
+			map[string]interface{}{"provider": provider, "service": service, "region": region.Id, "strategy": region.Data.Versions.Strategy})
+	}
+}
+
+func (scil *storeCloudInfoLoader) LoadImages(ctx context.Context, provider string, service string, region Region) {
+	switch region.Data.Images.Strategy {
+	case exact:
+		scil.log.Debug("loading images...")
+		scil.store.StoreImage(provider, service, region.Id, region.Data.Images.Data)
+		scil.log.Debug("images loaded")
+	case exclude:
+		var (
+			images interface{}
+			ok     bool
+		)
+		if images, ok = scil.store.GetImage(provider, scil.serviceData.Source, region.Id); !ok {
+			scil.log.Error("images not yet cached",
+				map[string]interface{}{"provider": provider, "service": scil.serviceData.Source, "region": region.Id})
+		} else {
+			var availableImages []cloudinfo.Image
+			var unavailableImage string
+			for _, image := range images.([]cloudinfo.Image) {
```

Comment:
This can be made a bit more readable:
```for _, image := range images.([]cloudinfo.Image) {
    keep := true
    for _, excludeImage := range region.Data.Images.Data {
        if image.Name == excludeImage.Name {
            keep = false
            break
        }
    }
    if keep {
        availableImages = append(availableImages, image)
    }
}```